File: problems/python_source_code/hard/182_b.py
N = int(input())
A = list(map(int, input().split()))

max_cnt = 0
ans = None
#kを全探索
for k in range(2, 1001):
    cnt = 0
    for a in A:
        if a % k == 0:
            cnt += 1

    if max_cnt < cnt:
        max_cnt = cnt 
        ans = k 

print(ans)

File: problems/cpp_source_code/hard/182_b.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  int  n;
  cin >> n;
  vector<int> a(n);
  for (int i = 0; i < n; i++) cin >> a[i];

  int ans = 0, mx = 0;
  for (int x = 2; x <= 1000; x++) {
    int cnt = 0;
    for (int i = 0; i < n; i++) {
      if (a[i] % x == 0) cnt++;
    }
    if (mx < cnt) {
      ans = x;
      mx = cnt;
    }
  }
  cout << ans << endl;
}